src/classes/CCON_PreventUpdate_TEST.cls

```diff
@@ -0,0 +1,207 @@
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2019
+* @group Course Enrollments
+* @group-content ../../ApexDocContent/CourseEnrollments.htm
+* @description Unit tests for CCON_PreventUpdate_TDTM. These tests
+* make sure Contact or Course Offering on a Course Connection do not
+* change when Term Grades or Attendance Events are assocaited it.
+*/
+
+
+@isTest
+private class CCON_PreventUpdate_TEST {
+    
+    @testSetup
+    static void dataSetup() {
+    
+        //Insert a contact
+        Contact con = UTIL_UnitTestData_TEST.getContact();
+        insert con;
+        
+        // Insert a Course Offering
+        Course_offering__c courseOffering = UTIL_UnitTestData_TEST.createCourseOffering(NULL, NULL);
+        
+        // Insert Course Connection
+        course_Enrollment__c courseEnrollment = UTIL_UnitTestData_TEST.getCourseConnection(con.Id, courseOffering.Id);
+        insert courseEnrollment;
+        
+    }
+    
+    // Test Contact and Course Offering updates with no related records on Course Connection 
+    
+    @isTest 
+    static void testUpdateCourseConnWithNoRelatedRecords() {
+    
+        //Insert a contact
+        Contact newCon = New Contact();
+        newCon.FirstName = 'TestNew01';
+        newCon.LastName = 'LastNew01';
+        newCon.Email = 'LastNew01@sfdo.com';
+        newCon.Phone = '0987123456';
+        insert newCon;
+        
+        // Insert a Course Offering
+        Course_offering__c newCourseOffering = UTIL_UnitTestData_TEST.createCourseOffering(NULL, NULL);
+        
+        course_Enrollment__c courseEnrollment = [SELECT Id, Contact__c, Course_Offering__c FROM Course_Enrollment__c LIMIT 1];
+        courseEnrollment.Contact__c = newCon.Id;
+        courseEnrollment.Course_offering__c = newCourseOffering.Id;
+        Test.startTest();
+            update courseEnrollment;
+        Test.stopTest();    
+        
+        System.assertEquals(courseEnrollment.Contact__c, newCon.Id);
+        System.assertEquals(courseEnrollment.Course_offering__c, newCourseOffering.Id);
+    
+    }
+    
+    // Test Contact and Course Offering updates with Term Grades related to the Course Connection 
+    
+    @isTest 
+    static void testUpdateCourseConnWithTermGrade() {
+    
+        course_Enrollment__c courseEnrollment = [SELECT Id, Contact__c, Course_Offering__c FROM Course_Enrollment__c LIMIT 1];
+        course_offering__c courseOffering = [SELECT Id FROM Course_Offering__c LIMIT 1];
+        Contact con = [SELECT Id FROM Contact LIMIT 1];
+        courseEnrollment.Course_Offering__c = courseOffering.Id;
+        update courseEnrollment;
+        
+        // Insert Term Grades
+        Term_Grade__c termGrade = new Term_Grade__c(Course_Connection__c = courseEnrollment.Id, Course_Offering__c = courseOffering.Id, Contact__c = con.Id);
```

Comment:
Could this be added to UTIL_UnitTestData_TEST class so we can reuse it elsewhere later if desired?